rmw_iceoryx_cpp/src/rmw_count.cpp

```diff
@@ -84,6 +84,46 @@ rmw_count_subscribers(
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+rmw_count_services(
+  const rmw_node_t * node,
+  const char * service_name,
+  size_t * count)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_ERROR);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(service_name, RMW_RET_ERROR);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(count, RMW_RET_ERROR);
+
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    rmw_count_services
+    : node, node->implementation_identifier,
+    rmw_get_implementation_identifier(), return RMW_RET_ERROR);
+
+  /// @todo There is no API to find out which 'ServiceDescription' is offered by which node
+  RMW_SET_ERROR_MSG("rmw_count_services is not supported in iceoryx");
```

Comment:
@mossmaurice Hmm, you can provide the node name with the client and server options. Do we have the problem that this needs to be included in the introspection topics? For publisher and subscriber, this is available, right?

Reply:
@budrus I don't think that's possible. How would one access the server options here in this function? 

> Do we have the problem that this needs to be included in the introspection topics?

Correct, there are currently no introspection topics available for client/server. Hence, for the ROS graph I use `service_discovery.findService(..)` and [provide a dummy name](https://github.com/ros2/rmw_iceoryx/blob/iron/rmw_iceoryx_cpp/src/internal/iceoryx_topic_names_and_types.cpp#L159-L180).

> For publisher and subscriber, this is available, right?

:heavy_check_mark: 

Two options would be possible:

1. Add client/server introspection topics
2. Extend `iox::runtime::ServiceDiscovery` to return us info about the node association

Reply:
@mossmaurice OK. 1 would be consistent with how we do it for pub/sub.

Reply:
@budrus I agree, we already have an issue for that https://github.com/eclipse-iceoryx/iceoryx/issues/1128